cmd/metacache-walk.go

```diff
@@ -126,6 +127,7 @@ func (s *xlStorage) WalkDir(ctx context.Context, opts WalkDirOptions, wr io.Writ
 
 	scanDir = func(current string) error {
 		// Skip forward, if requested...
+		var sb bytes.Buffer
```

Comment:
can we not use `strings.Builder` here? or better use github.com/valyala/bytebufferpool ? we already use this - looks like no harm in using it?

Reply:
`strings.Builder` allocates on each reset. This will keep the buffer and only allocate on on first use and final output (conversion to string). You are welcome to feed it a byte buffer as seed, but this already eliminate most allocs. This is simple and get us most of the way.